### PR TITLE
Add Package linter to validate path existence and url validity

### PR DESCRIPTION
**Pull request by adellibovi**
Resolves https://github.com/tuist/tuist/issues/2202

### Short description 📝

Solves issue #2202, this PR add a new lint for Project which validates:
* local path existence
* url syntax validity

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] A developer other than the author has verified that the changes work as expected.
- [x] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

Am I missing something else?

P.S.: @pepibumur my first contribution is my way of saying enjoy your new flat, some people give furniture, some other PR I guess :)

### CODE REVIEW

**Review by kwridan — COMMENTED** [review]
Nice work @adellibovi - thanks for submitting this!

Spotted a few minor things but this looks great overall.

File: Sources/TuistGenerator/Linter/PackageLinter.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class PackageLinter: PackageLinting {
+    private let fileHandler: FileHandling
+
+    init(fileHandler: FileHandling = FileHandler.shared) {
+        self.fileHandler = fileHandler
+    }
+
+    func lint(_ package: Package) -> [LintingIssue] {
+        if case let .local(path) = package, !fileHandler.exists(path) {
+            let issue = LintingIssue(reason: "Package with local path (\(path)) does not exists.",
```
[the remark this answers is not in the document]
Comment (adellibovi) [reply Sources/TuistGenerator/Linter/PackageLinter.swift:19]: Oops, nice catch!

File: Tests/TuistGeneratorTests/Linter/PackageLinterTests.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        super.setUp()
+        subject = PackageLinter()
+    }
+
+    override func tearDown() {
+        subject = nil
+        super.tearDown()
+    }
+
+    func test_lint_when_a_local_path_does_not_exists() {
```
[the remark this answers is not in the document]
Comment (adellibovi) [reply Tests/TuistGeneratorTests/Linter/PackageLinterTests.swift:23]: Thought about it and forgot hehe going to add right away :) thanks!

**Comment by adellibovi** [comment]
Thanks @kwridan for the fast review! 😄

**Review by kwridan — APPROVED** [review]
Thanks for the updates!

Tested locally ✅

Screenshot 2021-01-08 at 17 06 33
Screenshot 2021-01-08 at 16 59 51

One last bit, please update the change log to help highlight this new addition in the next release :)

**Comment by pepibumur** [comment]
@all-contributors add @adellibovi for code

**Comment by pepibumur** [comment]
Thanks for tackling this @adellibovi ✌️

**Review by pepibumur — APPROVED** [review]

**Pull request merged**